Production/plugins/TauTupleProducer.cc

```diff
@@ -950,6 +1097,51 @@ class TauTupleProducer : public edm::EDAnalyzer {
         }
     }
 
+    void FillSV(const std::vector<reco::VertexCompositePtrCandidate>*& SVts, const TauJet::PFCandCollection& cands)
+    {
+
+	auto findMatch = [&](const edm::Ptr<reco::Candidate>& recocand) {
+	    for(size_t cand_idx = 0; cand_idx < cands.size(); ++cand_idx) {
+                auto pfcand = dynamic_cast<const reco::Candidate*>(cands.at(cand_idx).candidate);
+		if(pfcand == &(*recocand))
+		   return static_cast<int>(cand_idx);
+	    }
+	    return -1;
+	};
+
+	auto collectCands = [&](const reco::VertexCompositePtrCandidate& sv) {
+	    std::set<int> cand_indices;
+	    for(size_t it = 0; it < sv.numberOfSourceCandidatePtrs(); it++) {
+                const edm::Ptr<reco::Candidate>& recocand = sv.sourceCandidatePtr(it);
+		int match_idx = findMatch(recocand);
+		if(match_idx >= 0) 
+		   cand_indices.insert(match_idx);
+	    }
+	    return cand_indices;
+	};
+
+        for(unsigned int sv_idx = 0; sv_idx < SVts->size(); ++sv_idx) {
+	   const auto& iSVtx = SVts->at(sv_idx);
+	   auto candsIdx = collectCands(iSVtx);
+	   if(candsIdx.size() > 0){
+	      tauTuple().sv_x.push_back(static_cast<float>(iSVtx.position().x()));
+              tauTuple().sv_y.push_back(static_cast<float>(iSVtx.position().y()));
+              tauTuple().sv_z.push_back(static_cast<float>(iSVtx.position().z()));
+              tauTuple().sv_t.push_back(static_cast<float>(iSVtx.t()));
+	      tauTuple().sv_xE.push_back(static_cast<float>(std::sqrt(iSVtx.vertexCovariance(0,0))));
+              tauTuple().sv_yE.push_back(static_cast<float>(std::sqrt(iSVtx.vertexCovariance(1,1))));
+              tauTuple().sv_zE.push_back(static_cast<float>(std::sqrt(iSVtx.vertexCovariance(2,2))));
+              tauTuple().sv_tE.push_back(static_cast<float>(iSVtx.tError()));
+	      tauTuple().sv_chi2.push_back(static_cast<float>(iSVtx.vertexChi2()));
+              tauTuple().sv_ndof.push_back(static_cast<float>(iSVtx.vertexNdof()));
+	      for(int cand_idx : candsIdx) {
+		 tauTuple().sv_cands_svIdx.push_back(sv_idx);
+		 tauTuple().sv_cands_candIdx.push_back(cand_idx);
```

Comment:
Currently only the indices of PF candidates which are assigned to `TauJet` object will be saved to `sv_cands_candIdx` (if they come from the associated SV). 

 Would it make sense to store all the PF indices which were used to reconstruct a given SV to have a more complete picture? For example, it could hint if we loose some candidates while applying the tau cone selection in `TauJet.cc`.

Reply:
yes, storing all candidates associated to SV makes sense. I think it would be better to include them at TauJet building stage to make sure that they are all stored in TauTuple.